StochasticBarrierFunctions/benchmarks/sostools_benchmarks/linear/contraction.m

```diff
@@ -0,0 +1,106 @@
+%% SOS: Contraction
+clc; clear all;
+
+deg_list = [2, 4, 8, 12, 24, 30];
+results = struct('degree', [], 'Bx', [], 'betaval', [], 'gam', [], 'Ps', [], 'time', []);
+
+for kk = 1:length(deg_list)
+    deg = deg_list(kk);
+    
+    % Start timer for this experiment
+    tic
+    [Bxpolys, betaval, gam, Ps] = runSOS2D(deg);
+    elapsedTime = toc;
+    
+    results(kk).degree = deg;
+    results(kk).Bx = Bxpolys;
+    results(kk).betaval = betaval;
+    results(kk).gam = gam;
+    results(kk).Ps = Ps;
+    results(kk).time = elapsedTime;
+    
+    fprintf('Degree: %d, gam = %.4f, betaval = %.4f, Ps = %.4f, time = %.2f s\n', ...
+        deg, gam, betaval, Ps, elapsedTime);
+end
+
+%% Function definition
+function [Bxpolys, betaval, gam, Ps] = runSOS2D(deg)
+    alpha = 1; gx = 0.1; N = 10; x0 = [0;0];
+    
+    syms z x1 x2 betasym gamsym real
```

Comment:
Instead of z, it should be z1 and z2. They are separate variables and should be evaluated separately (otherwise, the noise would be a diagonal line).